Add PSP to the deprecation guide (#26629)
kubernetes/website: content/en/docs/reference/using-api/deprecation-guide.md

* All existing persisted objects are accessible via the new API
* No notable changes

> #### PodSecurityPolicy {#psp-v125}

Reviewer: my initial version kept this page limited to things with clear replacements. I was going to see what shook out for PSP this release, and if cronjob graduates to v1 as expected in 1.21, update this section for both of them after that.

Reviewer: also, PSP deprecation is getting released with 1.21, so I'd hold this at least until then

Reviewer 2: I'm OK with merging this into the live website before v1.21 released, provided that:
- the deprecation in v1.21 is planned out, including how we communicate it
- we are confident beyond any reasonable doubt of the deprecation timeline, we are committing ourselves to make exactly the changes we're documenting here
- we make this update in tandem with the broader piece about communiction

Reviewer 2: See PR https://github.com/kubernetes/website/pull/27369, _Add blog post describing PSP deprecation and next steps_.

Reviewer 2: (now merged)

> PodSecurityPolicy in the **policy/v1beta1** API version will no longer be served in v1.25, and the PodSecurityPolicy admission controller will be removed.

Reviewer: ```suggestion
The **policy/v1beta1** API version of PodSecurityPolicy will no longer be served in v1.25, and the PodSecurityPolicy admission plugin will be removed.
```

Reviewer 2: Would be good to add this, but I don't see it blocking a merge.

> PodSecurityPolicy replacements are still under discussion, but current use can be migrated to

Reviewer: I would avoid putting uncertain future-looking statements in this doc

```suggestion
Current use can be migrated to
```

Reviewer 2: We can link to the blog article from https://github.com/kubernetes/website/pull/27369, which I believe is due to merge and publish _before_ the v1.21 release. Usually we don't link to blog articles for information about features but in this case I (confidently) think that's  a good idea.

> PodSecurityPolicy replacements are still under discussion, but current use can be migrated to

Reviewer 3: :wave: @tallclair .
I'm a bit surprised that this page includes information about future feature deprecations (about 1 yr away?).

nit: PodSecurityPolicy replacements are still under discussion; however, you can migrate current PodSecurityPolicy resources  to ...

Reviewer: the ones already included in the page are current deprecations (if you `kubectl get` those resources on a 1.20 API server, you get a message that they _are_ deprecated, when they'll be removed, and what you should switch to using)

Reviewer 2: > current deprecations

makes sense to me

We could merge this change at the same time as the deprecation announcement.

* Notable changes:
    * `spec.selector` is now required and immutable after creation; use the existing template labels as the selector for seamless upgrades

#### PodSecurityPolicy {#psp-v116}

The **extensions/v1beta1** API version of PodSecurityPolicy is no longer served as of v1.16.

> * Migrate manifests and API client to use the **policy/v1beta1** API version, available since v1.10.

Reviewer: ```suggestion
* Migrate manifests and API clients to use the **policy/v1beta1** API version, available since v1.10.
```

> * Note that the **policy/v1beta1** API version of PodSecurityPolicy will be removed in v1.25.

Reviewer: ```suggestion
* Note that the **policy/v1beta1** API version of PodSecurityPolicy is also deprecated and will be removed in v1.25.
```
